xbmc/pvr/PVRPlaybackState.h

```diff
@@ -0,0 +1,212 @@
+/*
+ *  Copyright (C) 2012-2019 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+class CDateTime;
+class CFileItem;
+
+namespace PVR
+{
+class CPVRChannel;
+class CPVRChannelGroup;
+class CPVREpgInfoTag;
+class CPVRRecording;
+
+class CPVRPlaybackState
+{
+public:
+  /*!
+   * @brief ctor.
+   */
+  CPVRPlaybackState();
+
+  /*!
+   * @brief dtor.
+   */
+  virtual ~CPVRPlaybackState();
+
+  /*!
+   * @brief Inform that playback of an item just started.
+   * @param item The item that started to play.
+   */
+  void OnPlaybackStarted(const std::shared_ptr<CFileItem> item);
+
+  /*!
+   * @brief Inform that playback of an item was stopped due to user interaction.
+   * @param item The item that stopped to play.
+   * @return True, if the state has changed, false otherwise
+   */
+  bool OnPlaybackStopped(const std::shared_ptr<CFileItem> item);
+
+  /*!
+   * @brief Inform that playback of an item has stopped without user interaction.
+   * @param item The item that ended to play.
+   */
+  void OnPlaybackEnded(const std::shared_ptr<CFileItem> item);
+
+  /*!
+   * @brief Check if a TV channel, radio channel or recording is playing.
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlaying() const;
+
+  /*!
+   * @brief Check if a TV channel is playing.
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlayingTV() const;
+
+  /*!
+   * @brief Check if a radio channel is playing.
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlayingRadio() const;
+
+  /*!
+   * @brief Check if a an encrypted TV or radio channel is playing.
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlayingEncryptedChannel() const;
+
+  /*!
+   * @brief Check if a recording is playing.
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlayingRecording() const;
+
+  /*!
+   * @brief Check if an epg tag is playing.
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlayingEpgTag() const;
+
+  /*!
+   * @brief Check whether playing channel matches given uids.
+   * @param iClientID The client id.
+   * @param iUniqueChannelID The channel uid.
+   * @return True on match, false if there is no match or no channel is playing.
+   */
+  bool IsPlayingChannel(int iClientID, int iUniqueChannelID) const;
+
+  /*!
+   * @brief Check if the given channel is playing.
+   * @param channel The channel to check.
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlayingChannel(const std::shared_ptr<CPVRChannel>& channel) const;
+
+  /*!
+   * @brief Check if the given recording is playing.
+   * @param recording The recording to check.
+   * @return True if it's playing, false otherwise.
+   */
+  bool IsPlayingRecording(const std::shared_ptr<CPVRRecording> &recording) const;
```

Comment:
> Left-align `*` and `&` to the base type they modify.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-pointer-and-reference-types

Reply:
Oversight on my end. Will change that.